codalab/server/worker_info_accessor.py

```diff
@@ -24,8 +26,19 @@ def worker_with_id(self, user_id, worker_id):
                 return worker
         return None
 
-    def user_owned_workers(self, user_id):
-        return list(self._user_id_to_workers[user_id])
+    def user_owned_workers(self, user_id, request_queue=None):
+        workers_list = self._user_id_to_workers[user_id]
```

Comment:
Apart from changing the shallow copy to deep copy, I also changed the logic of this function to be: 
1. If request_queue is specified for this bundle, the function will return all the personal workers that belong to this user_id.
2. If request_queue is not specified for this bundle, the function will return all the public workers.

Reply:
I find it a bit strange that this function does these two separate things. I think the filtering by tag belongs more to the filter function than here (and this function can stay a short, simple function)